### ISSUE TITLE
Modify Editor Config

### ISSUE THREAD

**Issue by mmstick** [kkawakam/rustyline#137]
The Ion sytsem shell requires the ability to toggle between Vi and Emacs modes, as well as history file parameters, on demand, but the current API requires reconstructing the entire Editor to change the config.

**Comment by mmstick** [comment]
@gwenn I'd expect for all methods available to `Config` to be able for the `Editor`, too. Exposed by a `Config` trait.

**Comment by gwenn** [comment]
@mmstick I am not sure but could you please give a try to:
https://github.com/gwenn/rustyline/tree/configurer
?
Thanks.

**Comment by mmstick** [comment]
I won't be able to try it out for a while. Ion relies very heavily on Liner at the moment.

**Comment by mmstick** [comment]
@gwenn Question, but does this have the PROMPT_SP workaround that fix issues with a missing newline in the last output? Have just recently fixed that in liner.

**Comment by gwenn** [comment]
@mmstick If you tell me how to reproduce the issue (with or without Ion), I will be able to investigate.

**Comment by mmstick** [comment]
@gwenn To test for that, you should be able to just write some text to stdout without an ending newline, and flush stdout so that it's written to the terminal. In the Ion shell, this is what happens now:

```
m/d/S/ion (pipe-exec-cleanup:8e2d7f|+30 -2) # echo -n one
one⏎                                                                                                      
m/d/S/ion (pipe-exec-cleanup:8e2d7f|+30 -2) # echo one
one
```

Without the workaround, the `one⏎` line would not be displayed, since the prompt would overwrite it. We also add `⏎` to the output to indicate that it was missing a newline. The workaround mostly involves writing as many spaces as the width of the terminal, minus the length of the omitted character, and then clearing the prompt. If done right, commands that properly ended with a newline will not have an extra newline, whereas those that lack the newline at the end will get displayed rather than erased.

**Comment by gwenn** [comment]
Should be fixed with 2.0.1 release
